sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -171,8 +171,26 @@ object ColumnPruning extends Rule[LogicalPlan] {
 
       Project(substitutedProjection, child)
 
+    case gen@Generate(_: Explode, _, _, _, c) =>
```

Comment:
SPARK-6489:this issue means it pruned columns when explode lateral view clause.
Does it need to extend to all generate clause? I want to mark generate node's parent reference in analysis phase. Is there another gracefully method to get  its parent references?